xbmc/GUIInfoManager.cpp

```diff
@@ -4990,8 +4990,6 @@ std::string CGUIInfoManager::GetItemLabel(const CFileItem *item, int info, std::
         else
           URIUtils::GetParentPath(item->GetVideoInfoTag()->m_strFileNameAndPath, path);
```

Comment:
Well, videodb items (files) will return the parent path. From a second look the entire logic is at best sub-optimal. For files node items we're currently returning the parent path whether or not it is a folder or file.